arquillian-webcam-video-recorder/src/main/java/org/arquillian/extension/recorder/video/webcam/configuration/WebcamVideoRecorderConfigurator.java

```diff
@@ -0,0 +1,127 @@
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.arquillian.extension.recorder.video.webcam.configuration;
+
+import org.arquillian.extension.recorder.video.*;
+import org.arquillian.extension.recorder.video.event.VideoExtensionConfigured;
+import org.arquillian.recorder.reporter.ReporterConfiguration;
+import org.jboss.arquillian.config.descriptor.api.ArquillianDescriptor;
+import org.jboss.arquillian.config.descriptor.api.ExtensionDef;
+import org.jboss.arquillian.core.api.Instance;
+import org.jboss.arquillian.core.api.InstanceProducer;
+import org.jboss.arquillian.core.api.annotation.ApplicationScoped;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.core.api.annotation.Observes;
+import org.jboss.arquillian.core.spi.ServiceLoader;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Observes:
+ * <ul>
+ * <li>{@link VideoExtensionConfigured}</li>
+ * </ul>
+ * Produces {@link ApplicationScoped}:
+ * <ul>
+ * <li>{@link VideoConfiguration}</li>
+ * <li>{@link VideoStrategy}</li>
+ * </ul>
+ *
+ * @author <a href="mailto:asotobu@gmail.com">Alex Soto</a>
+ */
+public class WebcamVideoRecorderConfigurator extends VideoConfigurator {
+
+    private static final Logger logger = Logger.getLogger(WebcamVideoRecorderConfigurator.class.getSimpleName());
+
+    @Inject
+    @ApplicationScoped
+    private InstanceProducer<VideoConfiguration> configuration;
+
+    @Inject
+    @ApplicationScoped
+    private InstanceProducer<VideoStrategy> strategy;
+
+    @Inject
+    private Instance<ServiceLoader> serviceLoader;
+
+    @Inject
+    private Instance<ReporterConfiguration> reporterConfiguration;
+
+    @Inject
+    private Instance<VideoRecorderEnvironmentCleaner> cleaner;
+
+    public void afterVideoExtensionConfigured(@Observes VideoExtensionConfigured event, ArquillianDescriptor descriptor) {
+        VideoConfiguration configuration = new WebcamVideoConfiguration(reporterConfiguration.get());
+
+        for (ExtensionDef extension : descriptor.getExtensions()) {
+            if (extension.getExtensionName().equals(EXTENSION_NAME)) {
+                configuration.setConfiguration(extension.getExtensionProperties());
+                configuration.validate();
```

Comment:
Yeah I know :) Maybe that should be repaired there as well. I did a lot of these classes and it kind of evolveed to the point that new techniques are not applied everywhere ...